astropy/table/tests/test_mixin.py

```diff
@@ -0,0 +1,344 @@
+    """
+    Assert all(t[name] == col), with special handling for known mixin cols.
+    """
+    if isinstance(col, coordinates.SkyCoord):
+        assert np.all(t[name].ra == col.ra)
+        assert np.all(t[name].dec == col.dec)
+    elif isinstance(col, u.Quantity):
+        if type(t) is QTable:
+            assert np.all(t[name].value == col.value)
+    elif isinstance(col, table_helpers.ArrayWrapper):
+        assert np.all(t[name].data == col.data)
+    else:
+        assert np.all(t[name] == col)
+
+def test_get_items(mixin_cols):
+    """
+    Test that slicing / indexing table gives right values and col attrs inherit
+    """
+    attrs = ('name', 'unit', 'dtype', 'format', 'description', 'meta')
+    m = mixin_cols['m']
+    col_setattr(m, 'name', 'm')
+    col_setattr(m, 'format', '{0}')
+    col_setattr(m, 'description', 'd')
+    col_setattr(m, 'meta', {'a': 1})
+    t = QTable([m])
+    for item in ([1, 3], np.array([0, 2]), slice(1, 3)):
+        t2 = t[item]
+        assert_table_name_col_equal(t2, 'm', m[item])
+        for attr in attrs:
+            assert col_getattr(t2['m'], attr) == col_getattr(m, attr)
+
+def test_add_column(mixin_cols):
+    """
+    Test that adding a column preserves values and attributes
+    """
+    attrs = ('name', 'unit', 'dtype', 'format', 'description', 'meta')
+    m = mixin_cols['m']
+    assert col_getattr(m, 'name') is None
+
+    # Make sure adding column in various ways doesn't touch
+    t = QTable([m], names=['a'])
+    assert col_getattr(m, 'name') is None
+
+    t['new'] = m
+    assert col_getattr(m, 'name') is None
+
+    col_setattr(m, 'name', 'm')
+    col_setattr(m, 'format', '{0}')
+    col_setattr(m, 'description', 'd')
+    col_setattr(m, 'meta', {'a': 1})
+    t = QTable([m])
+
+    # Add columns m2 and m3 by two different methods and test expected equality
+    t['m2'] = m
+    col_setattr(m, 'name', 'm3')
+    t.add_columns([m], copy=True)
+    col_setattr(m, 'name', 'm4')
+    t.add_columns([m], copy=False)
+    for name in ('m2', 'm3', 'm4'):
+        assert_table_name_col_equal(t, 'm', t[name])
+        for attr in attrs:
+            if attr != 'name':
+                assert col_getattr(t['m'], attr) == col_getattr(t[name], attr)
+
+def test_vstack():
+    """
+    Vstack tables with mixin cols.
+    """
+    t1 = QTable(MIXIN_COLS)
+    t2 = QTable(MIXIN_COLS)
+    with pytest.raises(NotImplementedError):
+        vstack([t1, t2])
+
+def test_insert_row(mixin_cols):
+    """
+    Test inserting a row, which only works for BaseColumn and Quantity
+    """
+    t = QTable(mixin_cols)
+    col_setattr(t['m'], 'description', 'd')
+    if isinstance(t['m'], u.Quantity):
+        t.insert_row(1, t[-1])
+        assert t[1] == t[-1]
+        assert col_getattr(t['m'], 'description') == 'd'
+    else:
+        with pytest.raises(ValueError) as exc:
+            t.insert_row(1, t[-1])
+        assert "Unable to insert row" in str(exc.value)
+
```

Comment:
Maybe you should also add a test to ensure that if you insert a row where one of the Quantities has inconsistent units, it behaves in an expected way? I noted there was a comment above about this, but don't see a test of the behavior (although maybe I missed it somewhere else?)